### ISSUE TITLE
 Avoid using shell commands for browser detection

### ISSUE THREAD

**Issue by PayBas** [DevExpress/testcafe-browser-tools#237]
```js
const MACOS_GET_BROWSER_LIST_COMMAND = 'ls "/Applications/" | grep -E "Chrome|Firefox|Opera|Safari|Chromium|Edge" | sed -E "s/ /032/"';
```
in https://github.com/DevExpress/testcafe-browser-tools/blob/master/src/api/get-installations.js#L14 has claimed another victim.

I've just spent an entire evening trying to figure out why TestCafe was throwing errors like:
```
    "  code: 'E1005',\n" +
    "  data: [ 'chrome:headless' ]\n" +
```

Eventually found the `testcafe --list-browsers` command, which returns no output! 
And then the arduous undertaking of trying to figure out what the hell was going on.

Turns out it's the aliases and functions in from https://ohmybash.nntoan.com/ that are making `MACOS_GET_BROWSER_LIST_COMMAND` unreliable for me.

So basically https://github.com/DevExpress/testcafe/issues/3821 again.

It has been opted that this functionality should be reworked in https://github.com/DevExpress/testcafe-browser-tools/issues/168 , but it was never actually implemented.

I strongly feel that relying on `ls`, `grep` and `sed` for this on macOS is a _bad_ idea. The whole Bash/Zsh GNU/BSD mess is still very much alive.

It's just way to easy to break stuff. And the end-user has absolutely no idea that the problem could be caused by shell profile settings like aliases. Why would he/she? No other nodejs app/tool works like that. The issue is totally counter-intuitive.

**Comment by yasinkocak** [comment]
Replace it with NodeJS File system (fs)

https://nodejs.org/api/fs.html

**Comment by PavelMor25** [comment]
Hello,

Thank you for bringing this issue to our attention. We are aware of it and agree that it needs to be addressed. We will add this task to our backlog, but we cannot give you exact time frames for its resolution. However, if you have any suggestions or solutions in mind, you're welcome to submit them by opening a pull request. We will review it, and if everything looks good, we'll accept it.

**Comment by PayBas** [comment]
This issue still persists.

**Comment by Bayheck** [comment]
Unfortunately, we cannot address this issue at the moment as our team is currently focused on other tasks. We may reassess the priority of this task in the future if we receive additional requests. You are welcome to submit a pull request with the described functionality, and we will review and accept it if it meets our standards.
